internal/services/postgres/postgresql_flexible_server_database_resource.go

```diff
@@ -61,11 +61,12 @@ func resourcePostgresqlFlexibleServerDatabase() *pluginsdk.Resource {
 			},
 
 			"collation": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     true,
-				ValidateFunc: validate.DatabaseCollation,
-				Default:      "en_US.utf8",
+				Type:             pluginsdk.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
@neil-yechenwei rather than doing this, can we update the validation to validate the values being expected by the API for this field (presumably `[a-z]{2}_[A-Z]{2}\.[a-z0-9]{1,}`?) - `suppress.CaseDifference` is deprecated and will be removed in the future, but validating the input will fix this problem without this issue.

Reply:
Updating the validation would bring breaking change. Maybe it's unexpected, right?

Reply:
@tombuildsstuff , seems updating the validation is a breaking change. Is it acceptable?

Reply:
If the API is requiring this casing, which is it by virtue of returning this, then we should be validating user input to ensure it matches these values - since otherwise users will see a diff as per #22675 (and so would have to make this change to the value of this field anyway) - so whilst _technically_ a breaking change, since users would need to match the casing being used from the API here to remove the diff (which we'd be validating against), so this isn't really a breaking change for our use-case.

Reply:
Got it. Service team confirmed that they not only support above format you mentioned (en_US.utf8) but also other formats like "en-US", "English_United States.1252". See more details from https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/postgres/validate/database_collation_test.go. And service team is also using the regex that is similar with https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/postgres/validate/database_collation.go since they also can't prevent the issue user mentioned. So does it make sense to close this issue and suggest user to use `ignore_changes` or specific value to workaround this issue?

Reply:
@neil-yechenwei it should be possible to obtain this list from Postgres and use that: https://database.guide/how-to-return-a-list-of-available-collations-in-postgresql/

Reply:
@tombuildsstuff , after checked with service team, we followed the link you above mentioned. We got a long list and there are a few different patterns. Seems it's very tough to extract a regex from them.

The part of the list:
```
kln-KE-x-icu
xh_ZA
zh_SG.utf8
nl_NL@euro
POSIX
C
default
ucs_basic
SQL_Latin1_General_CP1_CI_AS
SQL_AltDiction_CP850_CI_AI
English_United States.1252
utf8_general_ci
utf8_unicode_ci
En-US
Japanese_Bushu_Kakusu_100_CS_AS_KS_WS
utf8mb4_unicode_520_ci
```

Reply:
@neil-yechenwei if we have the complete list of values we can use that rather than a regex?

Reply:
@tombuildsstuff , though we could do this. But the collation list returned by https://database.guide/how-to-return-a-list-of-available-collations-in-postgresql/ is not fully collation list. For example, it doesn't include known collations "En-US" and "English_United States.1252". Given service team also can't provide the complete list, so service team is still using the rough validation.

Reply:
@neil-yechenwei as per [this article](https://database.guide/how-to-return-a-list-of-available-collations-in-postgresql/) we can obtain this list from a cluster once provisioned, so there's no need to involve the Service Team here.

As requested above, please run this command against a cluster to obtain the list of valid values, so that we can add those to a `validation.StringInSlice` function.

Reply:
@tombuildsstuff , I provisioned a cluster and I ran the command "select collname from pg_collation;". But I found the list returned by this command is incomplete list. So I assume we can't add those to a validation.StringInSlice function.

Reply:
Out of interest, can you post that list?

Reply:
[This one](https://gist.github.com/neil-yechenwei/82d2e0a953e1f961b9cfa196b32ff4d8) is for postgresql server db.
[Another one](https://gist.github.com/neil-yechenwei/561a070a518fb2ed664d914a18fa44b1) is for postgresql flexible server db.

Reply:
@neil-yechenwei cool, that list looks reasonable to validate against to me, we should be able to add that for now and then iterate on it, should other values be needed. We can also normalize the API response based on this list too, as we do in other places, to workaround this casing inconsistency - so can we update this?

Reply:
@tombuildsstuff , sure. Updated PR.